examples/lines_bars_and_markers/markevery_demo.py

```diff
@@ -3,99 +3,94 @@
 Markevery Demo
 ==============
 
-This example demonstrates the various options for showing a marker at a
-subset of data points using the ``markevery`` property of a Line2D object.
-
-Integer arguments are fairly intuitive.  e.g. ``markevery=5`` will plot every
-5th marker starting from the first data point.
-
-Float arguments allow markers to be spaced at approximately equal distances
-along the line.  The theoretical distance along the line between markers is
-determined by multiplying the display-coordinate distance of the axes
-bounding-box diagonal by the value of ``markevery``.  The data points closest
-to the theoretical distances will be shown.
-
-A slice or list/array can also be used with ``markevery`` to specify the
-markers to show.
+The ``markevery`` property of `.Line2D` allows drawing markers only at a
+subset of data points.
+
+The exact list of possible parameters is specified at `.Line2D.set_markevery`.
+In short:
+
+- A single integer N draws every N-th marker.
+- A tuple of integers (start, N) draws every N-th marker, starting at index
+  *start*.
+- A list of integers draws the markers at the specified indices.
+- A slice draws the markers at the sliced indices.
+- A float results in approximately even spacing along the curve. The distance
+  between markers is length of the Axes diagonal in data coordinates
+  multiplied by the given float.
```

Comment:
This is not your description below, and of course you can't have diagonal lengths in data co-ordinates because the two axes could have different units and wildly different dynamic ranges:
```suggestion
- A float results in approximately even spacing along the curve in physical space, and 
  is a fraction of the physical size of the axes diagonal on the screen.  
```
... is it true that this is what we do?  I wonder if this almost needs a separate figure, showing what happens if the same value, say 0.1, is used on two different sized axes....

Reply:
Reworded the whole bullet point to

```
- A float specifies the distance between markers as a fraction of the Axes
  diagonal in screen space. This will lead to a visually uniform distribution
  of the points along the line, irrespective of scales and zooming.
```